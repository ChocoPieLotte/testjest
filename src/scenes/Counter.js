import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View, Text, TouchableOpacity, StyleSheet, TextInput, Alert } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as counter from '../redux/actions/counter';
import { colors } from '../global';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  count: {
    fontWeight: 'bold',
    fontSize: 70
  },
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 6,
    width: 40,
    height: 40,
    margin: 8
  },
  buttonText: {
    color: 'black',
    fontSize: 24
  },
  input: {
    backgroundColor: 'gray',
    alignSelf: "center",
    width: '80%',
    height: 44
  }
});


export class Counter extends Component {
  static defaultProps = {
    count: 0
  };

  static propTypes = {
    decrement: PropTypes.func.isRequired,
    increment: PropTypes.func.isRequired,
    count: PropTypes.number
  };

  state = {
    text: ''
  };

  showBundle = () => Alert.alert(DeviceInfo.getBundleId());

  render() {
    const { count, decrement, increment } = this.props;
    return (
      <View style={styles.container}>
        <Text>{DeviceInfo.getDeviceId()}</Text>
        <TextInput
          style={styles.input}
          onChangeText={text => this.setState({ text })}
          value={this.state.text}
        />
        <Text style={styles.count}>{count}</Text>
        <View style={{ flexDirection: 'row' }}>
          <TouchableOpacity
            key={'key_decrement'}
            style={[styles.button, { backgroundColor: colors.flatRed }]}
            onPress={() => decrement()}
          >
            <Text style={styles.buttonText}>-</Text>
          </TouchableOpacity>
          <TouchableOpacity
            key={'key_increment'}
            style={[styles.button, { backgroundColor: colors.flatGreen }]}
            onPress={() => increment()}
          >
            <Text style={styles.buttonText}>+</Text>
          </TouchableOpacity>
        </View>
        <TouchableOpacity key={'key_bundle'} onPress={this.showBundle}>
          <Text>Show Bundle</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

export default connect(
  state => ({ count: state.counter.count }),
  dispatch => bindActionCreators({
    decrement: counter.decrement,
    increment: counter.increment
  }, dispatch)
)(Counter);
