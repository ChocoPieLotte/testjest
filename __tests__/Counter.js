import React from 'react';
import { Alert } from 'react-native';
import DeviceInfo from 'react-native-device-info'
import { Counter } from '../src/scenes/Counter';

import {shallow} from 'enzyme';


jest.mock('react-native-device-info', () => ({
  getDeviceId: jest.fn(() => 'deviceId'),
  getBundleId: jest.fn(() => 'com.fake.bundle')
}));

jest.mock('Alert', () => ({
  alert: jest.fn(),
}));

describe('Counter', () => {
  const decrement = jest.fn();
  const increment = jest.fn();
  const wrapper = shallow(<Counter decrement={decrement} increment={increment}/>);

  it('renders correctly', () => {
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should call decrement', () => {
    const button = wrapper.findWhere(n => n.key() === 'key_decrement');
    button.simulate('press');
    expect(decrement).toBeCalled();
  });

  it('should call increment', () => {
    const button = wrapper.findWhere(n => n.key() === 'key_increment');
    button.simulate('press');
    expect(increment).toBeCalled();
  });

  it('should input text and change state', () => {
    const input = wrapper.find('TextInput');
    input.simulate('ChangeText', 'sample text');
    expect(wrapper.state('text')).toEqual('sample text');
  });

  it('should call showBundle', () => {
    const spy = jest.spyOn(wrapper.instance(), 'showBundle');
    const button = wrapper.findWhere(n => n.key() === 'key_bundle');
    button.simulate('press');
    expect(spy).toBeCalled();
    expect(DeviceInfo.getBundleId).toBeCalled();
    expect(Alert.alert).toBeCalledWith('com.fake.bundle');
  });

});