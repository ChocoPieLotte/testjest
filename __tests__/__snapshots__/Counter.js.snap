// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      deviceId
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "gray",
          "height": 44,
          "width": "80%",
        }
      }
      value=""
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 70,
          "fontWeight": "bold",
        }
      }
    >
      0
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 6,
              "height": 40,
              "justifyContent": "center",
              "margin": 8,
              "width": 40,
            },
            Object {
              "backgroundColor": "#e74c3c",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontSize": 24,
            }
          }
        >
          -
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 6,
              "height": 40,
              "justifyContent": "center",
              "margin": 8,
              "width": 40,
            },
            Object {
              "backgroundColor": "#2ecc71",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontSize": 24,
            }
          }
        >
          +
        </Text>
      </TouchableOpacity>
    </View>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Show Bundle
      </Text>
    </TouchableOpacity>
  </View>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          deviceId
        </Text>,
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "gray",
              "height": 44,
              "width": "80%",
            }
          }
          value=""
        />,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 70,
              "fontWeight": "bold",
            }
          }
        >
          0
        </Text>,
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 6,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 8,
                  "width": 40,
                },
                Object {
                  "backgroundColor": "#e74c3c",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "black",
                  "fontSize": 24,
                }
              }
            >
              -
            </Text>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 6,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 8,
                  "width": 40,
                },
                Object {
                  "backgroundColor": "#2ecc71",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "black",
                  "fontSize": 24,
                }
              }
            >
              +
            </Text>
          </TouchableOpacity>
        </View>,
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Show Bundle
          </Text>
        </TouchableOpacity>,
      ],
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "deviceId",
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": "deviceId",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": true,
          "onChangeText": [Function],
          "style": Object {
            "alignSelf": "center",
            "backgroundColor": "gray",
            "height": 44,
            "width": "80%",
          },
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": 0,
          "ellipsizeMode": "tail",
          "style": Object {
            "fontSize": 70,
            "fontWeight": "bold",
          },
        },
        "ref": null,
        "rendered": 0,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 6,
                    "height": 40,
                    "justifyContent": "center",
                    "margin": 8,
                    "width": 40,
                  },
                  Object {
                    "backgroundColor": "#e74c3c",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "black",
                    "fontSize": 24,
                  }
                }
              >
                -
              </Text>
            </TouchableOpacity>,
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 6,
                    "height": 40,
                    "justifyContent": "center",
                    "margin": 8,
                    "width": 40,
                  },
                  Object {
                    "backgroundColor": "#2ecc71",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "black",
                    "fontSize": 24,
                  }
                }
              >
                +
              </Text>
            </TouchableOpacity>,
          ],
          "style": Object {
            "flexDirection": "row",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "key_decrement",
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "black",
                    "fontSize": 24,
                  }
                }
              >
                -
              </Text>,
              "onPress": [Function],
              "style": Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 6,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 8,
                  "width": 40,
                },
                Object {
                  "backgroundColor": "#e74c3c",
                },
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "-",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "black",
                  "fontSize": 24,
                },
              },
              "ref": null,
              "rendered": "-",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "key_increment",
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "black",
                    "fontSize": 24,
                  }
                }
              >
                +
              </Text>,
              "onPress": [Function],
              "style": Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 6,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 8,
                  "width": 40,
                },
                Object {
                  "backgroundColor": "#2ecc71",
                },
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "+",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "black",
                  "fontSize": 24,
                },
              },
              "ref": null,
              "rendered": "+",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "key_bundle",
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Show Bundle
          </Text>,
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Show Bundle",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "Show Bundle",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            deviceId
          </Text>,
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "gray",
                "height": 44,
                "width": "80%",
              }
            }
            value=""
          />,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 70,
                "fontWeight": "bold",
              }
            }
          >
            0
          </Text>,
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 6,
                    "height": 40,
                    "justifyContent": "center",
                    "margin": 8,
                    "width": 40,
                  },
                  Object {
                    "backgroundColor": "#e74c3c",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "black",
                    "fontSize": 24,
                  }
                }
              >
                -
              </Text>
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 6,
                    "height": 40,
                    "justifyContent": "center",
                    "margin": 8,
                    "width": 40,
                  },
                  Object {
                    "backgroundColor": "#2ecc71",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "black",
                    "fontSize": 24,
                  }
                }
              >
                +
              </Text>
            </TouchableOpacity>
          </View>,
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Show Bundle
            </Text>
          </TouchableOpacity>,
        ],
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "deviceId",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "deviceId",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": true,
            "onChangeText": [Function],
            "style": Object {
              "alignSelf": "center",
              "backgroundColor": "gray",
              "height": 44,
              "width": "80%",
            },
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": 0,
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 70,
              "fontWeight": "bold",
            },
          },
          "ref": null,
          "rendered": 0,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 6,
                      "height": 40,
                      "justifyContent": "center",
                      "margin": 8,
                      "width": 40,
                    },
                    Object {
                      "backgroundColor": "#e74c3c",
                    },
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "black",
                      "fontSize": 24,
                    }
                  }
                >
                  -
                </Text>
              </TouchableOpacity>,
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 6,
                      "height": 40,
                      "justifyContent": "center",
                      "margin": 8,
                      "width": 40,
                    },
                    Object {
                      "backgroundColor": "#2ecc71",
                    },
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "black",
                      "fontSize": 24,
                    }
                  }
                >
                  +
                </Text>
              </TouchableOpacity>,
            ],
            "style": Object {
              "flexDirection": "row",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "key_decrement",
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.2,
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "black",
                      "fontSize": 24,
                    }
                  }
                >
                  -
                </Text>,
                "onPress": [Function],
                "style": Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 6,
                    "height": 40,
                    "justifyContent": "center",
                    "margin": 8,
                    "width": 40,
                  },
                  Object {
                    "backgroundColor": "#e74c3c",
                  },
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "-",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "black",
                    "fontSize": 24,
                  },
                },
                "ref": null,
                "rendered": "-",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "key_increment",
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.2,
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "black",
                      "fontSize": 24,
                    }
                  }
                >
                  +
                </Text>,
                "onPress": [Function],
                "style": Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 6,
                    "height": 40,
                    "justifyContent": "center",
                    "margin": 8,
                    "width": 40,
                  },
                  Object {
                    "backgroundColor": "#2ecc71",
                  },
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "+",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "black",
                    "fontSize": 24,
                  },
                },
                "ref": null,
                "rendered": "+",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "key_bundle",
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Show Bundle
            </Text>,
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Show Bundle",
              "ellipsizeMode": "tail",
            },
            "ref": null,
            "rendered": "Show Bundle",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
